Vue web update (#412)
Picovoice/rhino: binding/vue/README.md

# Rhino Binding for Vue

## Rhino Speech-to-Intent engine

Made in Vancouver, Canada by [Picovoice](https://picovoice.ai)

Rhino is Picovoice's Speech-to-Intent engine. It directly infers intent from spoken commands within a given context of
interest, in real-time. For example, given a spoken command:

> Can I have a small double-shot espresso?
Rhino infers that the user would like to order a drink and emits the following inference result:

```json
{
  "isUnderstood": "true",
  "intent": "orderBeverage",
  "slots": {
    "beverage": "espresso",
    "size": "small",
    "numberOfShots": "2"
  }
}
```

Rhino is:

* using deep neural networks trained in real-world environments.
* compact and computationally-efficient, making it perfect for IoT.
* self-service. Developers and designers can train custom models using [Picovoice Console](https://console.picovoice.ai/).

## Compatibility

- Chrome / Edge
- Firefox

> - Safari

Reviewer: can you call this section `Browser Compatibility` and add a section called `Framework Compatibility` where you add:
```
- Vue.js 2.6.11+
- Vue.js 3.0.0+
```

```typescript
// Context (.rhn)
const rhinoContext = {
  publicPath: ${CONTEXT_RELATIVE_PATH},
  // or
  base64: ${CONTEXT_BASE64_STRING},
  // Optionals
  customWritePath: 'custom_context',
  forceWrite: true,
  version: 1,
  sensitivity: 0.5,
}
// Model (.pv)
const rhinoModel = {
  publicPath: ${MODEL_RELATIVE_PATH},
  // or
  base64: ${MODEL_BASE64_STRING},
  // Optionals
  customWritePath: 'custom_model',
  forceWrite: true,
  version: 1,
}
```

Additional engine options are provided via the `options` parameter.
Use `endpointDurationSec` and `requireEndpoint` to control the engine's endpointing behaviour.
An endpoint is a chunk of silence at the end of an utterance that marks the end of spoken command.

```typescript
// Optional. These are the default values
const options = {
  endpointDurationSec: 1.0,
  requireEndpoint: true,
}
```

### Initialize Rhino

**Note**: Due to limitations on Vue, one component can only have one instance of Rhino. To use multiple instances of Rhino instead, check out [Rhino Web binding](https://www.npmjs.com/package/@picovoice/rhino-web).

Create the following functions:

>  - `inferenceCallback` function to get the streaming results from the worker
>  - `contextCallback` function to get the context info from Rhino once the owrker is loaded
>  - `isLoadedCallback` function to check if Rhino has loaded
>  - `isListeningCallback` function to check if Rhino is listening for an inference
>  - `errorCallback` function to catch any error occurred

Reviewer: the wording here makes it sounds like the user has to call them to check:
```suggestion
 - `inferenceCallback` streams inference results from Rhino
 - `contextCallback` function that delivers context info when Rhino has completed initialization
 - `isLoadedCallback` function that executes when Rhino has loaded or unloaded
 - `isListeningCallback` function that executes when Rhino is listening for an inference
 - `errorCallback` function to catch any error occurred
```

```typescript
...
methods: {

>   keywordDetectionCallback: function(inference) {

Reviewer: ```suggestion
  inferenceCallback: function(inference) {
```

Once a conclusion is reached Rhino will enter a paused state. From the paused state Rhino call `process` again to detect another inference.

### Cleanup

When you are done with Rhino call `release`. This cleans up all resources used by Rhino and WebVoiceProcessor.

```typescript
await this.$rhino.release();
```

If any arguments require changes, call `release` then `init` again to initialize Rhino with the new settings.

## Contexts

Create custom contexts using the [Picovoice Console](https://console.picovoice.ai/).
Train the Rhino context model for the target platform WebAssembly (WASM).
Inside the downloaded `.zip` file, there will be a `.rhn` file which is the context model file in binary format.

Similar to the model file (`.pv`), keyword files (`.rhn`) are saved in IndexedDB to be used by Web Assembly.
Either `base64` or `publicPath` must be set to instantiate Rhino. If both are set, Rhino will use
the `base64` model.

## Non-English Languages

In order to detect non-English inferences you need to use the corresponding model file (`.pv`). The model files for all
supported languages are available [here](https://github.com/Picovoice/rhino/tree/master/lib/common).

## Demo

> For example usage refer to our [Web demo application](https://github.com/Picovoice/rhino/tree/master/demo/web).

Reviewer: ```suggestion
For example usage, refer to our [Vue demo application](https://github.com/Picovoice/rhino/tree/master/demo/vue).
```

## Installation

Using `yarn`:

```console
yarn add @picovoice/rhino-vue @picovoice/web-voice-processor
```

or using `npm`:

```console
npm install --save @picovoice/rhino-vue @picovoice/web-voice-processor
```

> ### AccessKey

Reviewer: Duplicate AccessKey section (you may want to double check your other bindings for this)
